hook!-full-game/src/Player/Player.gd

```diff
@@ -8,17 +8,23 @@ onready var floor_detector: RayCast2D = $FloorDetector
 onready var skin: Position2D = $Skin
 onready var skills: Node = $Skills
 onready var stats: Stats = $Stats
+onready var animation_player : AnimationPlayer = $AnimationPlayer
 
 var _info_dict: = {} setget _set_info_dict
 
 var _gravity: = 0.0
 var _velocity: = Vector2.ZERO
 
 var _state: = "idle"
+var checkpoints := []
```

Comment:
Checkpoints shouldn't be stored on the player node. This breaks single responsibility: why would the player script manage checkpoints? There should be a node in the scene tree to take care of checkpoints (e.g. a `Checkpoints` node) without the player's need to be aware of them.